pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/PurgeMinionClusterIntegrationTest.java

```diff
@@ -0,0 +1,380 @@
+    List<SegmentZKMetadata> segmentsZKMetadataDeltaPassed = _pinotHelixResourceManager
+        .getSegmentsZKMetadata(tablenameOfflinePassed);
+
+    Map<String, String> customSegmentMetadataPassed = new HashMap<>();
+    customSegmentMetadataPassed.put(MinionConstants.PurgeTask.TASK_TYPE
+        + MinionConstants.TASK_TIME_SUFFIX, String.valueOf(System.currentTimeMillis() - 88400000));
+
+    for (SegmentZKMetadata segmentZKMetadata : segmentsZKMetadataDeltaPassed) {
+      segmentZKMetadata.setCustomMap(customSegmentMetadataPassed);
+      _pinotHelixResourceManager.updateZkMetadata(tablenameOfflinePassed, segmentZKMetadata);
+    }
+    //set up not passed delay
+    List<SegmentZKMetadata> segmentsZKMetadataDeltaNotPassed = _pinotHelixResourceManager
+        .getSegmentsZKMetadata(tablenameOfflineNotPassed);
+    Map<String, String> customSegmentMetadataNotPassed = new HashMap<>();
+    customSegmentMetadataNotPassed.put(MinionConstants.PurgeTask.TASK_TYPE
+        + MinionConstants.TASK_TIME_SUFFIX, String.valueOf(System.currentTimeMillis() - 4000));
+    for (SegmentZKMetadata segmentZKMetadata : segmentsZKMetadataDeltaNotPassed) {
+      segmentZKMetadata.setCustomMap(customSegmentMetadataNotPassed);
+      _pinotHelixResourceManager.updateZkMetadata(tablenameOfflineNotPassed, segmentZKMetadata);
+    }
+  }
+
+  private TableConfig createOfflineTableConfig(String tableName, TableTaskConfig taskConfig) {
+    return createOfflineTableConfig(tableName, taskConfig, null);
+  }
+
+  private TableConfig createOfflineTableConfig(String tableName, TableTaskConfig taskConfig,
+      @Nullable SegmentPartitionConfig partitionConfig) {
+    return new TableConfigBuilder(TableType.OFFLINE).setTableName(tableName).setSchemaName(getSchemaName())
+        .setTimeColumnName(getTimeColumnName()).setSortedColumn(getSortedColumn())
+        .setInvertedIndexColumns(getInvertedIndexColumns()).setNoDictionaryColumns(getNoDictionaryColumns())
+        .setRangeIndexColumns(getRangeIndexColumns()).setBloomFilterColumns(getBloomFilterColumns())
+        .setFieldConfigList(getFieldConfigs()).setNumReplicas(getNumReplicas()).setSegmentVersion(getSegmentVersion())
+        .setLoadMode(getLoadMode()).setTaskConfig(taskConfig).setBrokerTenant(getBrokerTenant())
+        .setServerTenant(getServerTenant()).setIngestionConfig(getIngestionConfig())
+        .setNullHandlingEnabled(getNullHandlingEnabled()).setSegmentPartitionConfig(partitionConfig).build();
+  }
+
+  private TableTaskConfig getPurgeTaskConfig() {
+    Map<String, String> tableTaskConfigs = new HashMap<>();
+    tableTaskConfigs.put("deltaPurgeTimePeriod", "1d");
+    return new TableTaskConfig(Collections.singletonMap(MinionConstants.PurgeTask.TASK_TYPE, tableTaskConfigs));
+  }
+
+
+
+  private static void buildSegmentsFromAvroWithPurgeTime(List<File> avroFiles, TableConfig tableConfig,
+      Schema schema, int baseSegmentIndex, File segmentDir, File tarDir, String purgeTime)
```

Comment:
`purgeTime` is not used when creating the segments. You may create the segments once, and upload them to 3 tables